flamegraph.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+# To enable the required permissions, execute the following sequence as root:
+# $ chown root:wheel /proc/modules
+# $ chown root:wheel /proc/kallsyms
+# $ echo 0 > /proc/sys/kernel/kptr_restrict
+# $ echo -1 > /proc/sys/kernel/perf_event_paranoid
+#
+# Usage:
+#
+# Profile an executable
+# ./flamegraph.sh <executable>
+#
+# To change the frequency:
+# HZ=100 ./flamegraph.sh <executable>
+#
+# To use the maximum frequency
+# HZ=max ./flamegraph.sh <executable>
+#
+# This can be useful to see how much time is spent in carp vs clang
+# ./flamegraph.sh ./carp.sh -b <file.carp>
+
+HZ=${HZ:-1000}
+prg=`basename $1`
+svg=`mktemp XXXXXX.svg`
+perf record -F $HZ -a -g -- $*
+perf script | stackcollapse-perf.pl | flamegraph.pl > $svg
+xdg-open $svg
+rm $svg
```

Comment:
This will likely remove the svg before the view application had a change to open it.

Reply:
Oh, right. It seems to work for me though. 

Is there some way to view the svg sinchronously?

Reply:
Maybe use `read -p "Press enter to delete the temporary svg file" some_var`?

Reply:
Does `xdg-open` exit before reading the file?

Reply:
Not that I like it, but perhaps a `sleep 5` will be less intrusive than prompting...

Reply:
> Does xdg-open exit before reading the file?

`xdg-open` runs the program to open a file and then it exits. It doesn't wait for the program to exit.